From 69292a616808ac145ec9cd3a4e548ee7a6c87a81 Mon Sep 17 00:00:00 2001
From: Vincenzo Eduardo Padulano <vincenzo.eduardo.padulano@cern.ch>
Date: Wed, 29 May 2024 16:12:22 +0200
Subject: [PATCH] [core] Link clad to clang only if target available

The target is not available if we are not building clang in the ROOT build, e.g. in the ROOT conda releases.
---
 core/metacling/src/CMakeLists.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/core/metacling/src/CMakeLists.txt b/core/metacling/src/CMakeLists.txt
index 194d334cb9..eea33c6471 100644
--- a/core/metacling/src/CMakeLists.txt
+++ b/core/metacling/src/CMakeLists.txt
@@ -103,10 +103,12 @@ if (clad)
   else()
     set(CLING_PLUGIN_LINK_LIBS -Wl,--whole-archive cladPlugin cladDifferentiator -Wl,--no-whole-archive)
   endif()
-  # Link our clad libraries to clang. If users use the clang from ROOT they will
-  # also be able to use clad out of the box.
-  add_dependencies(clang clad)
-  target_link_libraries(clang PUBLIC ${CLING_PLUGIN_LINK_LIBS})
+  if(TARGET clang)
+    # Link our clad libraries to clang. If users use the clang from ROOT they will
+    # also be able to use clad out of the box.
+    add_dependencies(clang clad)
+    target_link_libraries(clang PUBLIC ${CLING_PLUGIN_LINK_LIBS})
+  endif()
 endif()
 
 ROOT_LINKER_LIBRARY(Cling
-- 
2.44.0

